app/helpers/mission_control/jobs/jobs_helper.rb

```diff
@@ -57,7 +57,7 @@ def as_renderable_hash(argument)
       elsif argument["_aj_serialized"] == "ActiveJob::Serializers::ModuleSerializer"
         argument["value"]
       else
-        ActiveJob::Arguments.deserialize([ argument ])
+        argument.transform_values { |value| as_renderable_argument(value) }
```

Comment:
I think this would prevent other serialized arguments from being deserialized. For example, with this change, a time argument looks like this: 

<img width="507" alt="Screenshot 2024-10-31 at 21 11 46" src="https://github.com/user-attachments/assets/3ac27047-ff31-4da1-86b9-2e95154ad4a9">

Reply:
Oh interesting, so there can be "hash" shaped job arguments that are actually serialized instances of an object? I didn't realize that... I will need to play around with this again to see how to differentiate these. I thought I was only affecting hash/keyword arguments here.

Reply:
Ah okay right, that's why I initially proposed a narrower fix that only affects keyword arguments. https://github.com/rails/mission_control-jobs/compare/main...ibrahima:handle-aj-keyword-arguments?expand=1

But I realized now the same issue applies to hash arguments. And also, if we do call `as_renderable_argument` here instead of using `ActiveJob::Arguments.deserialize()`, then keyword arguments come out looking a little funny.

```
 gid://dummy/Model/1, {"options"=>{"user"=>"gid://dummy/User/1", "_aj_symbol_keys"=>"(user)"}, "_aj_ruby2_keywords"=>"(options)"} 
```

whereas without this change, it reformats the symbol keys nicely (albeit while deserializing globalID arguments).

```
 gid://dummy/Model/1, {:options=>{:user=>#<User id: 1, ...>}} 
```

I feel like given this, the ideal scenario for me would be to add an argument to ActiveJob::Arguments.deserialize to optionally disable globalID deserialization, because it seems like we generally do want to use the hash deserialization logic from there, but the hash deserialization [also deserializes globalID arguments which we don't want in this case](https://github.com/rails/rails/blob/d0dcb8fa6073a0c4d42600c15e82e3bb386b27d3/activejob/lib/active_job/arguments.rb#L148). The narrower change of only affecting keyword arguments does work but it makes the keyword arguments look uglier unfortunately. And I don't see a good way to solve this in MissionControl::Jobs without copying a bunch of the code from ActiveJob::Arguments here and stripping out the part that does globalID deserialization, which seems silly.

Do you think it would be worthwhile to do https://github.com/rails/mission_control-jobs/compare/main...ibrahima:handle-aj-keyword-arguments?expand=1 even though it makes the keyword arguments look worse (with all the metadata keys visible) or would it be better to solve it without that drawback?

Reply:
> Do you think it would be worthwhile to do https://github.com/rails/mission_control-jobs/compare/main...ibrahima:handle-aj-keyword-arguments?expand=1 even though it makes the keyword arguments look worse (with all the metadata keys visible) or would it be better to solve it without that drawback?

I think using keyword arguments that aren't whose values aren't global IDs is a more common case than using global IDs as values, so I wouldn't want to make that case worse to fix a less common one. 

I think to make progress here we should have a test first, with different arguments and the results we expect to see for each, instead of trying to catch those by trial and error.

Reply:
That's a good idea, fair point. Lately in our codebases we started using keyword arguments for almost everything because the newer Ruby syntax makes it only an extra colon compared to regular arguments if your variable name matches. But I actually don't think it makes sense to overuse it like that anyway.